dbt/include/bigquery/macros/materializations/incremental.sql

```diff
@@ -23,12 +23,13 @@
 
 {% endmacro %}
 {% macro bq_create_table_as(is_time_ingestion_partitioning, temporary, relation, compiled_code, language='sql') %}
-  {% if is_time_ingestion_partitioning %}
+  {% if is_time_ingestion_partitioning and language == 'sql' %}
```

Comment:
we should probably raise an explicit exception on unsupported configurations
```suggestion
  {% if is_time_ingestion_partitioning and language == 'python' %}
    {% do exceptions.raise_compiler_error(
      "Python models do not support ingestion time partitioning"
    ) %}
  {% endif %}
  {% if is_time_ingestion_partitioning and language == 'sql' %}
```